tool-plugins/intellij/src/main/java/io/ballerina/plugins/idea/codeinsight/daemon/impl/BallerinaAnnotator.java

```diff
@@ -70,9 +70,20 @@ public void annotate(@NotNull PsiElement element, @NotNull AnnotationHolder hold
                     Annotation annotation = holder.createInfoAnnotation(element, null);
                     annotation.setTextAttributes(BallerinaSyntaxHighlightingColors.ANNOTATION);
                 }
+                // these keywords can be used as iterable operations as well
+            } else if (elementType == BallerinaTypes.MAP || elementType == BallerinaTypes.FOREACH) {
+                if (parent.getNode().getElementType() == BallerinaTypes.RESERVED_WORD) {
+                    //if the context is an iterable operation, skips highlighting as a keyword
+                    if (parent.getParent() != null
+                            && parent.getParent().getNode().getElementType() == BallerinaTypes.ANY_IDENTIFIER_NAME) {
+                        annotateKeyword(element, holder, BallerinaSyntaxHighlightingColors.DEFAULT, false);
+                    }
+                } else {
+                    annotateKeyword(element, holder, false);
```

Comment:
We don't need to highlight the keyword in the else since it shouldn't be highlighted.

Reply:
**if** condition is used to check whether the context is an iterable operation. so if not, these keywords should be highlighted as keywords.